tmt/steps/provision/testcloud.py

```diff
@@ -836,9 +829,17 @@ def prepare_config(self) -> None:
         self.config.DOWNLOAD_PROGRESS = self.debug_level > 2
         self.config.DOWNLOAD_PROGRESS_VERBOSE = False
 
+        # We can't assign the path to STORE_DIR because it must exist first
+        data_dir = self.workdir_root / 'testcloud'
+        store_dir = data_dir / 'images'
```

Comment:
Oh, I just realized these should not be local variables, but properties of `self`! See `store_dir = self.workdir_root / 'testcloud/images'` below, it's constructing the same path, and we should avoid that and provide canonical "path where images are stored" attribute. So, how about the following?

```python
@functools.cached_property
def testcloud_data_dirpath(self) -> Path:
    return self.workdir_root / 'testcloud'

@functools.cached_property
def testcloud_image_dirpath(self) -> Path:
    return self.testcloud_data_dirpath / 'testcloud'
```

And then use these, `os.makedirs(self.testcloud_data_dirpath, exist_ok=True)` and so on.

Reply:
@happz can we resolve?